app/src/main/java/org/dosomething/letsdothis/ui/RegisterActivity.java

```diff
@@ -45,10 +63,148 @@ public void onCreate(Bundle savedInstanceState)
+        File externalFile = new File(getExternalFilesDir(Environment.DIRECTORY_PICTURES),
+                                     "avatar.jpg");
+        imageUri = Uri.fromFile(externalFile);
+        if(BuildConfig.DEBUG)
+        {
+            Log.d("photo location", imageUri.toString());
+        }
+        takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
+
+        String pickTitle = getString(R.string.select_picture);
+        Intent chooserIntent = Intent.createChooser(takePhotoIntent, pickTitle);
+        chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, new Intent[] {pickIntent});
+
+        startActivityForResult(chooserIntent, SELECT_PICTURE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data)
+    {
+        if(resultCode == RESULT_OK)
+        {
+            if(requestCode == SELECT_PICTURE)
+            {
+                final boolean isCamera;
+                if(data == null)
+                {
+                    isCamera = true;
+                }
+                else
+                {
+                    final String action = data.getAction();
+                    if(action == null)
+                    {
+                        isCamera = false;
+                    }
+                    else
+                    {
+                        isCamera = action.equals(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+                    }
+                }
+
+                Uri selectedImageUri;
+                if(isCamera)
+                {
+                    selectedImageUri = imageUri;
+                }
+                else
+                {
+                    selectedImageUri = data.getData();
+                }
+                if(BuildConfig.DEBUG)
+                {
+                    Log.d("test-----------", selectedImageUri.toString());
+                }
+
+                Picasso.with(this).load(selectedImageUri).resize(avatar.getWidth(), avatar.getHeight()).into(avatar);
+
+                File externalFile = new File(getExternalFilesDir(Environment.DIRECTORY_PICTURES),
+                                             "avatar.jpg");
+                saveFile(selectedImageUri, externalFile);
+                //FIXME: this should be loaded into Hub
+                AppPrefs.getInstance(this).setAvatarPath(externalFile.getAbsolutePath());
+            }
+        }
+    }
+
+    private void saveFile(Uri sourceUri, File externalFile)
+    {
+
+        final int chunkSize = 1024;
+        byte[] imageData = new byte[chunkSize];
+
+        InputStream in = null;
+        OutputStream out = null;
+        try
+        {
+            in = getContentResolver().openInputStream(sourceUri);
+            out = new FileOutputStream(externalFile);
+
+            int bytesRead;
+            while((bytesRead = in.read(imageData)) > 0)
+            {
+                out.write(Arrays.copyOfRange(imageData, 0, Math.max(0, bytesRead)));
```

Comment:
copy to temp file and then rename. you can get a temp file in the same dir by naming it "temp" + System.currentTimeMillis() + ".png"